### PR TITLE
Support users to customize JsonUtil

### PR DESCRIPTION
**Pull request by oxsean**
## What is the purpose of the change
In http scenario, the user wants to customize the JSON conversion.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by CrazyHZM — CHANGES_REQUESTED** [review]
I can't entirely agree with this `JsonUtilCustomizer.` It increases the complexity of jsonutil and allows users to re-implement jsonutil if they want to customize some parameters

**Comment by oxsean** [comment]
> I can't entirely agree with this `JsonUtilCustomizer.` It increases the complexity of jsonutil and allows users to re-implement jsonutil if they want to customize some parameters

It's really something we need to consider, This feature is for rest, the complexity is relative, for the end user to use `JsonUtilCustomizer ` will make the customization easier, otherwise, because fastjson all use static parameters, customization is hard.

File: dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/HttpJsonUtils.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.apache.dubbo.rpc.Constants;
+import org.apache.dubbo.rpc.model.FrameworkModel;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+
+public final class HttpJsonUtils {
+
+    private static JsonUtil jsonUtil;
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/HttpJsonUtils.java]: Prevent use static here

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by CrazyHZM — APPROVED** [review]

**Comment by CrazyHZM** [comment]
@oxsean The pr title could be changed.

**Comment by AlbumenJ** [comment]
@chickenlj PTAL

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**